INSTALL.md

```diff
@@ -1,95 +1,508 @@
+- If agbcc has **not been built before** in the folder where you chose to store pokeemerald, run the following commands to build and install it into pokeemerald:
+
+    ```bash
+    git clone https://github.com/luckytyphlosion/agbcc
+    cd agbcc
+    ./build.sh
+    ./install.sh ../pokeemerald
+    ```
+
+- **Otherwise**, if agbcc has been built before (e.g. if the git clone above fails), but was **last built on a different terminal** than the one currently used (only relevant to Windows, e.g. switching from msys2 to WSL1), then run the following commands to build and install it into pokeemerald:
+
+    ```bash
+    cd agbcc
+    git clean -fX
+    ./build.sh
+    ./install.sh ../pokeemerald
+    ```
+
+- **Otherwise**, run the following commands to install agbcc into pokeemerald:
+
+    ```bash
+    cd agbcc
+    ./install.sh ../pokeemerald
+    ```
+
+    > Note: If building agbcc or pokeemerald results in an error, try deleting the agbcc folder and re-installing agbcc as if it has not been built before.
+
+3. Once agbcc is installed, change directory back to the base directory where pokeemerald and agbcc are stored:
+
+    ```bash
+    cd ..
+    ```
+
+Now you're ready to [build **pokeemerald**](#build-pokeemerald)
+## Build pokeemerald
+If you aren't in the pokeemerald directory already, then **change directory** to the pokeemerald folder:
+```bash
+cd pokeemerald
+```
+To build **pokeemerald.gba** for the first time and confirm it matches the official ROM image (Note: to speed up builds, see [Parallel builds](#parallel-builds)):
+```bash
+make compare
+```
+If an OK is returned, then the installation went smoothly.
+> Note for Windows: if you switched terminals since the last build (e.g. from msys2 to WSL1), you must run `make clean-tools` once before any subsequent `make` commands.
+
+To build **pokeemerald.gba** with your changes:
+```bash
+make
+```
+
+# Building guidance
 
 ## Parallel builds
 
 See [the GNU docs](https://www.gnu.org/software/make/manual/html_node/Parallel.html) and [this Stack Exchange thread](https://unix.stackexchange.com/questions/208568) for more information.
 
-To speed up building, run:
-
-	make -j$(nproc)
+To speed up building, first get the value of `nproc` by running the following command:
+```bash
+nproc
+```
+Builds can then be sped up by running the following command:
+```bash
+make -j<output of nproc>
+```
+Replace `<output of nproc>` with the number that the `nproc` command returned.
 
 `nproc` is not available on macOS. The alternative is `sysctl -n hw.ncpu` ([relevant Stack Overflow thread](https://stackoverflow.com/questions/1715580)).
 
-
 ## Debug info
 
 To build **pokeemerald.elf** with enhanced debug info:
+```bash
+make DINFO=1
+```
 
-	make DINFO=1
+## devkitARM's C compiler
 
+This project supports the `arm-none-eabi-gcc` compiler included with devkitARM. If devkitARM (a.k.a. gba-dev) has already been installed as part of the platform-specific instructions, simply run:
```

Comment:
Should installing `gcc-arm-none-eabi` be listed as an alternative here? Its size isn't going to be relevant to everyone, and it only takes a single command to install

Reply:
I suppose, but it isn't a necessity and the INSTALL.md has dragged on long enough. I might do it in the further future when I have more time.